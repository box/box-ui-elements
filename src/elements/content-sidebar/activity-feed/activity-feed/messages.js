/**
 * @flow
 * @file i18n messages
 * @author Box
 */

import { defineMessages } from 'react-intl';

const messages = defineMessages({
    annotationMissingError: {
        id: 'be.activitySidebar.activityFeed.annotationMissingError',
        defaultMessage: 'This comment no longer exists',
        description: 'Text to show when an annotation activity no longer exists',
    },
    feedInlineErrorTitle: {
        id: 'be.activitySidebar.activityFeed.feedInlineErrorTitle',
        defaultMessage: 'Error',
        description: 'Error title',
    },
    taskMissingError: {
        id: 'be.activitySidebar.activityFeed.taskMissingError',
        defaultMessage: 'This task no longer exists',
        description: 'Text to show when a task no longer exists',
    },
    commentMissingError: {
        id: 'be.activitySidebar.activityFeed.commentMissingError',
        defaultMessage: 'This comment no longer exists',
        description: 'Text to show when comment no longer exists',
    },
});

export default messages;
