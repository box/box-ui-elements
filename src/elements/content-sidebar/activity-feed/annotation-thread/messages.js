import { defineMessages } from 'react-intl';

const messages = defineMessages({
    errorFetchAnnotation: {
        id: 'be.annotattionThread.errorFetchAnnotation',
        description: 'Error message when an annotation fetch fails',
        defaultMessage: 'The annotation could not be fetched.',
    },
    errorEditAnnotation: {
        id: 'be.annotationThread.errorEditAnnotation',
        description: 'Error message when an annotation update fails',
        defaultMessage: 'This annotation could not be modified.',
    },
    errorDeleteAnnotation: {
        id: 'be.annotationThread.errorDeleteAnnotation',
        description: 'Error message when an annotation deletion fails',
        defaultMessage: 'There was an error deleting this item.',
    },
});

export default messages;
