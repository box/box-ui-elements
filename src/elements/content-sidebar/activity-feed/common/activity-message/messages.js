// @flow
import { defineMessages } from 'react-intl';

const messages = defineMessages({
    activityMessageShowOriginal: {
        id: 'be.contentSidebar.activityFeed.commmon.showOriginalMessage',
        defaultMessage: 'Show Original',
        description: 'Show original button for showing original comment',
    },
    activityMessageTranslate: {
        id: 'be.contentSidebar.activityFeed.common.translateMessage',
        defaultMessage: 'Translate',
        description: 'Translate button for translating comment',
    },
});

export default messages;
