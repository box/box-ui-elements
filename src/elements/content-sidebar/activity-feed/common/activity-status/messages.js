// @flow
import { defineMessages } from 'react-intl';

const messages = defineMessages({
    activityStatusResolved: {
        id: 'be.contentSidebar.activityFeed.common.activityStatusResolved',
        defaultMessage: 'Resolved',
        description: 'Resolved status of Comment or Annotation visible in Feed Item',
    },
});

export default messages;
