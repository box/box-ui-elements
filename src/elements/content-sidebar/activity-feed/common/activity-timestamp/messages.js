// @flow
import { defineMessages } from 'react-intl';

const messages = defineMessages({
    fullDateTime: {
        id: 'be.activityFeed.fullDateTime',
        defaultMessage: '{time, date, full} at {time, time, short}',
        description: 'Full date time and time stamp',
    },
});

export default messages;
