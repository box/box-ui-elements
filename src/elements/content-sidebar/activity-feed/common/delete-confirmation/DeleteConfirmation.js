/**
 * @flow
 * @file Comment component
 */

import * as React from 'react';
import { FormattedMessage, type MessageDescriptor } from 'react-intl';

import Button from '../../../../../components/button';
import commonMessages from '../../../../common/messages';
import PrimaryButton from '../../../../../components/primary-button';
import { KEYS } from '../../../../../constants';
import { Overlay } from '../../../../../components/flyout';
import { ACTIVITY_TARGETS } from '../../../../common/interactionTargets';
import './DeleteConfirmation.scss';

type Props = {
    isOpen: boolean,
    message: MessageDescriptor,
    onDeleteCancel: Function,
    onDeleteConfirm: Function,
};

class DeleteConfirmation extends React.Component<Props> {
    onKeyDown = (event: SyntheticKeyboardEvent<>): void => {
        const { nativeEvent } = event;
        const { isOpen, onDeleteCancel } = this.props;

        nativeEvent.stopImmediatePropagation();

        switch (event.key) {
            case KEYS.escape:
                event.stopPropagation();
                event.preventDefault();
                if (isOpen) {
                    onDeleteCancel();
                }
                break;
            default:
                break;
        }
    };

    render() {
        const { isOpen, message, onDeleteCancel, onDeleteConfirm, ...rest } = this.props;

        return (
            <Overlay
                className="be-modal bcs-DeleteConfirmation"
                onKeyDown={this.onKeyDown}
                role="dialog"
                shouldDefaultFocus
                shouldOutlineFocus={false}
                {...rest}
            >
                <div className="bcs-DeleteConfirmation-promptMessage">
                    <FormattedMessage {...message} />
                </div>
                <div>
                    <Button
                        className="bcs-DeleteConfirmation-cancel"
                        onClick={onDeleteCancel}
                        type="button"
                        data-resin-target={ACTIVITY_TARGETS.INLINE_DELETE_CANCEL}
                    >
                        <FormattedMessage {...commonMessages.cancel} />
                    </Button>
                    <PrimaryButton
                        className="bcs-DeleteConfirmation-delete"
                        onClick={onDeleteConfirm}
                        type="button"
                        data-resin-target={ACTIVITY_TARGETS.INLINE_DELETE_CONFIRM}
                    >
                        <FormattedMessage {...commonMessages.delete} />
                    </PrimaryButton>
                </div>
            </Overlay>
        );
    }
}

export default DeleteConfirmation;
