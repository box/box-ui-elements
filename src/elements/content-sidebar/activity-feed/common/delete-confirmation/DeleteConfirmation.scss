@import '../../../../../styles/variables';
@import '../../../../../styles/mixins/overlay';

.bcs-DeleteConfirmation {
    width: 212px;

    .overlay {
        @include bdl-Overlay-container;

        padding: 20px;
    }
}

.bcs-DeleteConfirmation-promptMessage {
    margin-bottom: 20px;
}

// overrides for confirm/cancel button spacing
.btn.bcs-DeleteConfirmation-delete,
.btn.bcs-DeleteConfirmation-cancel {
    margin-left: 0;
}

.btn.bcs-DeleteConfirmation-cancel {
    margin-right: 10px;
}
