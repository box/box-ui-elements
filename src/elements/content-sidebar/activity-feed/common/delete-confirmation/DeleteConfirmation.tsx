import * as React from 'react';
import classNames from 'classnames';
import { FormattedMessage, MessageDescriptor, injectIntl } from 'react-intl';
import type { WrappedComponentProps } from 'react-intl';

import Button, { ButtonType } from '../../../../../components/button';
import commonMessages from '../../../../common/messages';
import PrimaryButton from '../../../../../components/primary-button';

import { Overlay } from '../../../../../components/flyout';
import { ACTIVITY_TARGETS } from '../../../../common/interactionTargets';
import './DeleteConfirmation.scss';

interface Props {
    /** Additional CSS class name */
    className?: string;
    /** Whether the confirmation dialog is shown */
    isOpen: boolean;
    /** Message to display in the confirmation dialog */
    message: MessageDescriptor;
    /** Callback when cancel button is clicked */
    onDeleteCancel: () => void;
    /** Callback when confirm button is clicked */
    onDeleteConfirm: () => void;
}

type DeleteConfirmationProps = Props & WrappedComponentProps;

class DeleteConfirmation extends React.Component<DeleteConfirmationProps> {
    render(): React.ReactElement | null {
        const { className, intl, isOpen, message, onDeleteCancel, onDeleteConfirm } = this.props;

        if (!isOpen) {
            return null;
        }

        return (
            <Overlay
                className={classNames('be-modal bcs-DeleteConfirmation', className)}
                role="dialog"
                shouldDefaultFocus
                shouldOutlineFocus={false}
                data-testid="delete-confirmation"
                onClose={onDeleteCancel}
            >
                <div className="bcs-DeleteConfirmation-promptMessage">
                    <FormattedMessage {...message} />
                </div>
                <div>
                    <Button
                        aria-label={intl.formatMessage(commonMessages.cancel)}
                        className="bcs-DeleteConfirmation-cancel"
                        onClick={onDeleteCancel}
                        type={ButtonType.BUTTON}
                        data-resin-target={ACTIVITY_TARGETS.INLINE_DELETE_CANCEL}
                    >
                        <FormattedMessage {...commonMessages.cancel} />
                    </Button>
                    <PrimaryButton
                        aria-label={intl.formatMessage(commonMessages.delete)}
                        className="bcs-DeleteConfirmation-delete"
                        onClick={onDeleteConfirm}
                        type={ButtonType.BUTTON}
                        data-resin-target={ACTIVITY_TARGETS.INLINE_DELETE_CONFIRM}
                    >
                        <FormattedMessage {...commonMessages.delete} />
                    </PrimaryButton>
                </div>
            </Overlay>
        );
    }
}

export { DeleteConfirmation as DeleteConfirmationBase };
export default injectIntl(DeleteConfirmation);
