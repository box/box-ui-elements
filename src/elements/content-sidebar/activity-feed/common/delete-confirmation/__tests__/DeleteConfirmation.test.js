import * as React from 'react';
import { mount } from 'enzyme';

import { DeleteConfirmationBase as DeleteConfirmation } from '../DeleteConfirmation';
import messages from '../../../comment/messages';

describe('elements/content-sidebar/ActivityFeed/common/delete-confirmation', () => {
    const getWrapper = props =>
        mount(
            <DeleteConfirmation
                isOpen
                message={messages.commentDeletePrompt}
                onDeleteCancel={jest.fn()}
                onDeleteConfirm={jest.fn()}
                intl={{ formatMessage: jest.fn() }}
                {...props}
            />,
        );

    describe('render()', () => {
        test('should render component', () => {
            const wrapper = getWrapper();
            expect(wrapper.find('.bcs-DeleteConfirmation-promptMessage')).toMatchSnapshot();
        });
    });

    describe('onKeyDown', () => {
        test('should handle Escape key', () => {
            const onDeleteCancelMock = jest.fn();
            const onDeleteConfirmMock = jest.fn();
            const wrapper = getWrapper({ onDeleteCancel: onDeleteCancelMock, onDeleteConfirm: onDeleteConfirmMock });
            const overlay = wrapper.find('Overlay');
            overlay.prop('onKeyDown')({ key: 'Escape', preventDefault: jest.fn(), stopPropagation: jest.fn() });
            expect(onDeleteCancelMock).toBeCalled();
            expect(onDeleteConfirmMock).not.toBeCalled();
        });
    });
});
