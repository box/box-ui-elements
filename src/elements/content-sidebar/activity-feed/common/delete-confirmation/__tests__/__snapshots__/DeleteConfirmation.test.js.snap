// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/content-sidebar/ActivityFeed/common/delete-confirmation render() should render component 1`] = `
<div
  className="bcs-DeleteConfirmation-promptMessage"
>
  <FormattedMessage
    defaultMessage="Are you sure you want to permanently delete this comment?"
    id="be.contentSidebar.activityFeed.comment.commentDeletePrompt"
  />
</div>
`;
